File: java/unidad_01/Problema_1_05/src/Problema_1_05.java
import java.util.Scanner;
import java.util.InputMismatchException;

/*
 * El programa, dado como datos el radio y la altura de un
 * cilindro, calcula su área y su volumen.
 */

public class Problema_1_05 {

    public static float calcularArea(float radio, float altura) {

        return 2 * (float) Math.PI * radio * altura;
        // Área de un cilindro
    }

    public static float calcularVolumen(float radio, float altura) {

        return (float) Math.PI * (float) Math.pow(radio, 2) * altura;
        // Volumen de un cilindro
    }

    public static void main(String[] args) {

        float radio, altura;

        try (Scanner scanner = new Scanner(System.in)) {
            /*
             * Es una buena práctica cerrar los recursos adecuadamente después de usarlos
             * para liberar cualquier recurso asociado, como archivos abiertos, conexiones
             * de red o lectores de entrada.
             */

            System.out.print("Radio: ");
            radio = scanner.nextFloat();

            System.out.print("Altura: ");
            altura = scanner.nextFloat();

            if (radio <= 0 || altura <= 0) {

                System.out.println("Error: el radio y la altura deben ser mayores que cero.");
                return;
            }
        } catch (InputMismatchException e) {
            // Se produce cuando el tipo de dato ingresado por el usuario no coincide con el
            // esperado.

            System.out.println(e);
            return;
        }
        float area = calcularArea(radio, altura);

        float volumen = calcularVolumen(radio, altura);

        System.out.printf("El área del cilindro es de %.2f u^2 y %n", area);

        System.out.printf("el volumen es de %.2f u^3", volumen);
    }
}

File: python/unidad_01/problema_1_05.py
import math

'''
El programa, dado como datos el radio y la altura de un
cilindro, calcula su área y su volumen.
'''

def calcular_area(radio, altura):

    return 2 * math.pi * radio * altura
    # Área de un cilindro

def calcular_volumen(radio, altura):

    return math.pi * radio ** 2 * altura
    # Volumen de un cilindro

def main():

    try:

        radio = float(input("Radio: "))

        altura = float(input("Altura: "))

        if radio <= 0 or altura <= 0:

            print("Error: el radio y la altura deben ser mayores que cero.")
            return
    
    except ValueError as e:
        # Se produce cuando una función recibe un argumento 
        # con un valor que no es válido para esa función.

        print(e)
        return

    area = calcular_area(radio, altura)

    volumen = calcular_volumen(radio, altura)

    print(f"El área del cilindro es de {area:.2f} u^2 y")
    
    print(f"el volumen es de {volumen:.2f} u^3")

if __name__ == '__main__':

    main()